modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -773,6 +881,51 @@ namespace cv {
                             setParams.setLayerBlobs(cv_layers_counter, bn_blobs);
                         }
                     }
+                    if (layer_type == "connected")
+                    {
+                        int output = getParam<int>(layer_params, "output", 1);
+                        bool use_batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+
+                        CV_Assert(output>0);
+                        size_t const weights_size = current_total * output;
+                        int sizes_weights[] = { output, current_total };
+                        cv::Mat weightsBlob;
+                        weightsBlob.create(2, sizes_weights, CV_32F);
+                        CV_Assert(weightsBlob.isContinuous());
+
+                        cv::Mat meanData_mat(1, output, CV_32F);	// mean
+                        cv::Mat stdData_mat(1, output, CV_32F);	// variance
+                        cv::Mat weightsData_mat(1, output, CV_32F);// scale
+                        cv::Mat biasData_mat(1, output, CV_32F);	// bias
+
+                        ifile.read(reinterpret_cast<char *>(biasData_mat.ptr<float>()), sizeof(float)*output);
+                        if (use_batch_normalize) {
+                            ifile.read(reinterpret_cast<char *>(weightsData_mat.ptr<float>()), sizeof(float)*output);
+                            ifile.read(reinterpret_cast<char *>(meanData_mat.ptr<float>()), sizeof(float)*output);
+                            ifile.read(reinterpret_cast<char *>(stdData_mat.ptr<float>()), sizeof(float)*output);
+                        }
+                        ifile.read(reinterpret_cast<char *>(weightsBlob.ptr<float>()), sizeof(float)*weights_size);
```

Comment:
Please move it to separate function due there are multiple entries with this BN layer

Reply:
Done !